### PR TITLE
docs: bump actions to latest major

### PR DESCRIPTION
**Pull request by crazy-max**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,16 +143,22 @@ jobs:
...
+service account based authentication.
 
 #### Workload identity federation based authentication
 
-Configure the workload identity federation for github actions in gcloud (for steps, [refer here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)). In the steps, your service account should the ability to push to GCR. Then use google-github-actions/auth action for authentication using workload identity like below:
+Configure the workload identity federation for github actions in gcloud (for
+steps, [refer here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation)).
+In the steps, your service account should the ability to push to GCR. Then use
+`google-github-actions/auth` action for authentication using workload identity
+like below:
```
Comment (dvdksn) [inline README.md:161]:
```suggestion
Configure the workload identity federation for GitHub Actions in Google Cloud,
[see here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation).
Your service account must have permission to push to GCR.
Use the `google-github-actions/auth` action to authenticate using workload identity
as shown in the following example:
```

File: README.md [hunk 2]
```diff
@@ -56,8 +57,8 @@ jobs:
 ### GitHub Container Registry
 
 To authenticate against the [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry),
```
Comment (dvdksn) [inline README.md:59]:
```suggestion
To authenticate to the [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry),
```

File: README.md [hunk 3]
```diff
@@ -161,33 +171,36 @@ jobs:
...
+      uses: docker/login-action@v3
       with:
         registry: gcr.io
         username: oauth2accesstoken
         password: ${{ steps.auth.outputs.access_token }}
 ```
 
-> Replace `<workload_identity_provider>` with configured workload identity provider. For steps to configure, [refer here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation).
+> Replace `<workload_identity_provider>` with configured workload identity
+> provider. For steps to configure, [refer here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation).
```
Comment (dvdksn) [inline README.md:192]:
```suggestion
> provider. For steps to configure, [see here](https://github.com/google-github-actions/auth#setting-up-workload-identity-federation).
```

File: README.md [hunk 4]
```diff
@@ -161,33 +171,36 @@ jobs:
...
 
-> Replace `<service_account>` with configured service account in workload identity provider which has access to push to GCR
+> Replace `<service_account>` with configured service account in workload
+> identity provider which has access to push to GCR
 
 #### Service account based authentication
 
 Use a service account with the ability to push to GCR and [configure access control](https://cloud.google.com/container-registry/docs/access-control).
```
Comment (dvdksn) [inline README.md:199]:
```suggestion
Use a service account with permission to push to GCR and [configure access control](https://cloud.google.com/container-registry/docs/access-control).
```

File: README.md [hunk 5]
```diff
@@ -161,33 +171,36 @@ jobs:
...
-Then create and download the JSON key for this service account and save content of `.json` file
-[as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
-called `GCR_JSON_KEY` in your GitHub repo. Ensure you set the username to `_json_key`,
-or `_json_key_base64` if you use a base64-encoded key.
+Then create and download the JSON key for this service account and save content
+of `.json` file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
+called `GCR_JSON_KEY` in your GitHub repo. Ensure you set the username to
+`_json_key`, or `_json_key_base64` if you use a base64-encoded key.
```
Comment (dvdksn) [inline README.md:203]:
```suggestion
Download the key for the service account as a JSON file.
Save the contents of the file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
named `GCR_JSON_KEY` in your GitHub repository.
Set the username to `_json_key`, or `_json_key_base64` if you use a base64-encoded key.
```

File: README.md [hunk 6]
```diff
@@ -211,11 +224,16 @@ jobs:
 
 ### Google Artifact Registry (GAR)
 
-You can use either workload identity federation based keyless authentication or  service account based authentication.
+You can use either workload identity federation based keyless authentication or
+service account based authentication.
```
Comment (dvdksn) [inline README.md:228]:
```suggestion
You can authenticate with workload identity federation or a service account.
```

File: README.md [hunk 7]
```diff
@@ -211,11 +224,16 @@ jobs:
...
 
 #### Workload identity federation based authentication
```
Comment (dvdksn) [inline README.md:230]:
```suggestion
#### Workload identity federation
```

File: README.md [hunk 8]
```diff
@@ -228,35 +246,39 @@ jobs:
...
 
 Use a service account with the ability to push to GAR and [configure access control](https://cloud.google.com/artifact-registry/docs/access-control).
-Then create and download the JSON key for this service account and save content of `.json` file
-[as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
-called `GAR_JSON_KEY` in your GitHub repo. Ensure you set the username to `_json_key`,
-or `_json_key_base64` if you use a base64-encoded key.
+Then create and download the JSON key for this service account and save content
+of `.json` file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
+called `GAR_JSON_KEY` in your GitHub repo. Ensure you set the username to
+`_json_key`, or `_json_key_base64` if you use a base64-encoded key.
```
Comment (dvdksn) [inline README.md:281]:
```suggestion
Download the key for the service account as a JSON file.
Save the contents of the file [as a secret](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
named `GCR_JSON_KEY` in your GitHub repository.
Set the username to `_json_key`, or `_json_key_base64` if you use a base64-encoded key.
```

File: README.md [hunk 9]
```diff
@@ -228,35 +246,39 @@ jobs:
...
+
+> Replace `<service_account>` with configured service account in workload
+> identity provider which has access to push to GCR
 
 > Replace `<location>` with the regional or multi-regional [location](https://cloud.google.com/artifact-registry/docs/repo-organize#locations)
 > of the repository where the image is stored.
 
 #### Service account based authentication
 
 Use a service account with the ability to push to GAR and [configure access control](https://cloud.google.com/artifact-registry/docs/access-control).
```
Comment (dvdksn) [inline README.md:277]:
```suggestion
Use a service account with permission to push to GAR and [configure access control](https://cloud.google.com/artifact-registry/docs/access-control).
```

File: README.md [hunk 10]
```diff
@@ -367,7 +389,8 @@ jobs:
 ### AWS Public Elastic Container Registry (ECR)
 
 Use an IAM user with the ability to [push to ECR Public with `AmazonElasticContainerRegistryPublicPowerUser` managed policy for example](https://docs.aws.amazon.com/AmazonECR/latest/public/public-ecr-managed-policies.html#AmazonElasticContainerRegistryPublicPowerUser).
```
Comment (dvdksn) [inline README.md:391]: Seems like the link was broken

```suggestion
Use an IAM user with permission to push to ECR Public, for example using [managed policies](https://docs.aws.amazon.com/AmazonECR/latest/userguide/security-iam-awsmanpol.html#security-iam-awsmanpol-AmazonEC2ContainerRegistryPowerUser).
```

File: README.md [hunk 11]
```diff
@@ -367,7 +389,8 @@ jobs:
...
-Then create and download access keys and save `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` [as secrets](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
+Then create and download access keys and save `AWS_ACCESS_KEY_ID` and
```
Comment (dvdksn) [inline README.md:392]:
```suggestion
Download the access keys and save them as `AWS_ACCESS_KEY_ID` and
```

File: README.md [hunk 12]
```diff
@@ -367,7 +389,8 @@ jobs:
...
+`AWS_SECRET_ACCESS_KEY` [as secrets](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
```
Comment (dvdksn) [inline README.md:393]:
```suggestion
`AWS_SECRET_ACCESS_KEY` [secrets](https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository)
```

File: README.md [hunk 13]
```diff
@@ -367,7 +389,8 @@ jobs:
...
 in your GitHub repo.
```
Comment (dvdksn) [inline README.md:394]:
```suggestion
in your GitHub repository.
```

File: README.md [hunk 14]
```diff
@@ -428,7 +451,8 @@ jobs:
 
 ### Quay.io
 
-Use a [Robot account](https://docs.quay.io/glossary/robot-accounts.html) with the ability to push to a public/private Quay.io repository.
+Use a [Robot account](https://docs.quay.io/glossary/robot-accounts.html) with
+the ability to push to a public/private Quay.io repository.
```
Comment (dvdksn) [inline README.md:455]:
```suggestion
permission to push to a Quay.io repository.
```

File: README.md [hunk 15]
```diff
@@ -454,15 +478,15 @@ jobs:
 
 ### inputs
 
-Following inputs can be used as `step.with` keys
+The following inputs can be used as `step.with` keys
```
Comment (dvdksn) [inline README.md:481]:
```suggestion
The following inputs can be used as `step.with` keys:
```

File: README.md [hunk 16]
```diff
@@ -454,15 +478,15 @@ jobs:
...
-| `password`       | String  |                             | Password or personal access token used to log against the Docker registry |
-| `ecr`            | String  | `auto`                      | Specifies whether the given registry is ECR (`auto`, `true` or `false`) |
-| `logout`         | Bool    | `true`                      | Log out from the Docker registry at the end of a job |
+| Name       | Type   | Default | Description                                                                   |
+|------------|--------|---------|-------------------------------------------------------------------------------|
+| `registry` | String |         | Server address of Docker registry. If not set then will default to Docker Hub |
+| `username` | String |         | Username used to log against the Docker registry                              |
+| `password` | String |         | Password or personal access token used to log against the Docker registry     |
+| `ecr`      | String | `auto`  | Specifies whether the given registry is ECR (`auto`, `true` or `false`)       |
+| `logout`   | Bool   | `true`  | Log out from the Docker registry at the end of a job                          |
```
Comment (dvdksn) [inline README.md:489]:
```suggestion
| `registry` | String |         | Server address of Docker registry. If not set then will default to Docker Hub |
| `username` | String |         | Username for authenticating to the Docker registry                              |
| `password` | String |         | Password or personal access for authenticating to the Docker registry     |
| `ecr`      | String | `auto`  | Specifies whether the given registry is ECR (`auto`, `true` or `false`)       |
| `logout`   | Bool   | `true`  | Log out from the Docker registry at the end of a job                          |
```

**Review by dvdksn — CHANGES_REQUESTED** [review]

**Comment by crazy-max** [comment]
@dvdksn Thanks for the review! Should be good now

**Review by dvdksn — APPROVED** [review]

**Pull request merged**